ddtrace/tracer.py

```diff
@@ -283,15 +283,21 @@ def configure(
                 # get the URL from.
                 url = None
 
+            if filters is None:
+                filters = self.writer._filters if hasattr(self, "writer") else []
+
             if hasattr(self, "writer") and self.writer.is_alive():
                 self.writer.stop()
 
-            self.writer = AgentWriter(
-                url,
-                sampler=self.sampler,
-                priority_sampler=self.priority_sampler,
-                dogstatsd=get_dogstatsd_client(self._dogstatsd_url),
-                report_metrics=config.health_metrics_enabled,
+            self.writer = TraceProcessor(
```

Comment:
Given that there are both a `writer` and a `processor` modules now, I think `self.writer` becomes a misleading name because it now holds a reference to a processor rather than an actual writer. So, if the writer is now becoming an implementation detail, would it make sense to rename this to `self.processor` instead? I would then suggest that the processor implements a `run` method rather than `write`.

Reply:
Totally agreed. I don't think that we can do away with `tracer.writer` as I believe we have recommended to people to do `tracer.writer.flush_queue()` in the past. We could expose `writer` as a property that does `self.processor.writer` to deal with this though and would actually maintain the functionality better.

I'll make that change thanks @P403n1x87!